app/components/add-validations-to-form.js

```diff
@@ -46,10 +49,19 @@ export default class AddValidationsToFormComponent extends Component {
 
     this.builderStore.registerObserver(() => {
       this.serializeToTtlCode(this.builderStore);
-    }, REGISTERED_FORM_TTL_CODE_KEY);
+    }, OBSERVER_KEY.validationForm);
   }
 
   serializeToTtlCode(builderStore) {
+    if (this.args.isComponentNotShown) {
```

Comment:
Wait, isn't the real issue that we don't deregister when the component is being destroyed (when switching tabs) I think using the [willDestroy](https://api.emberjs.com/ember/release/modules/@glimmer%2Fcomponent#willdestroy) hook would be a better place to do that.

If the early return here is still needed after that I think you can also just check the `this.isDestroying` or `this.isDestroyed` properties.

Reply:
I was looking for a hook but only found [these](https://guides.emberjs.com/v2.14.0/components/the-component-lifecycle/) and they where never called when testing this with console logs.. this willwillDestroy hook does the same and is indeed better

Reply:
Yea, Ember 2.14 is _very_ old 😄. Those are the docs for the old "ember component" which are different from the Glimmer components we are using now.

Reply:
Damned always check the version indeed and than the type of component used is also important ..